File: iPlant.FMS.Models/Structs/elg/ELGCatalog.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace iPlant.FMS.Models
{
    public class ELGCatalog
    {
        /**
             * 目录ID
             */
        public int CatalogID { get; set; } = 0;

        /**
         * 目录或文件名称
         */
        public String CatalogName { get; set; } = "";

        /**
         * 是否是目录
         */
        public Boolean IsCatalog { get; set; } = true;

        /**
         * 文件路径
         */
        public String FilePath { get; set; } = "";

        public DateTime UpdateTime { get; set; } = new DateTime(2000, 1, 1);


        public long Length { get; set; } = 0;

        /**
         * 目录的子项集合
         */
        public List<ELGCatalog> SubCatalogList { get; set; } = new List<ELGCatalog>();


    }
}

File: iPlant.FMS.Models/Structs/elg/ELGCatalog.java
package com.mes.server.service.po.elg;

import java.io.Serializable;
import java.util.List;
import java.util.List;

public class ELGCatalog implements Serializable {
	/**
	 * 序列号
	 */
	private static final long serialVersionUID = 1L;

	/**
	 * 目录ID
	 */
	public int CatalogID;

	/**
	 * 目录或文件名称
	 */
	public String CatalogName = "";

	/**
	 * 是否是目录
	 */
	public boolean IsCatalog = true;

	/**
	 * 文件路径
	 */
	public String FilePath = "";

	/**
	 * 目录的子项集合
	 */
	public List<ELGCatalog> SubCatalogList = new List<ELGCatalog>();

	public String getFilePath() {
		return FilePath;
	}

	public void setFilePath(String filePath) {
		FilePath = filePath;
	}

	public int getCatalogID() {
		return CatalogID;
	}

	public void setCatalogID(int catalogID) {
		CatalogID = catalogID;
	}

	public String getCatalogName() {
		return CatalogName;
	}

	public void setCatalogName(String catalogName) {
		CatalogName = catalogName;
	}

	public boolean isIsCatalog() {
		return IsCatalog;
	}

	public void setIsCatalog(boolean isCatalog) {
		IsCatalog = isCatalog;
	}

	public List<ELGCatalog> getwSubCatalogList() {
		return SubCatalogList;
	}

	public void setwSubCatalogList(List<ELGCatalog> wSubCatalogList) {
		this.SubCatalogList = wSubCatalogList;
	}
}
